pyaedt/modeler/stackup_3d.py

```diff
@@ -0,0 +1,2284 @@
+    >>> wave_length_formula = "c0/" + my_frequency.name
+    >>> my_wave_length = NamedVariable(hfss, "my_wave_length", wave_length_formula)
+    >>> my_permittivity = NamedVariable(hfss, "my_permittivity", "2.2")
+    >>> my_wave_length.expression = my_wave_length.expression + "/" + my_permittivity.name
+
+    """
+
+    def __init__(self, application, name, expression):
+        self._application = application
+        self._name = name
+        self._expression = expression
+        application[name] = expression
+
+    @property
+    def _variable(self):
+        return self._application.variable_manager.variables[self._name]
+
+    @property
+    def name(self):
+        """Return the name of the variable as a string."""
+        return self._name
+
+    @property
+    def expression(self):
+        """Return the expression of the variable as a string."""
+        return self._expression
+
+    @expression.setter
+    def expression(self, expression):
+        """Set the expression of the variable.
+        Parameters
+        ----------
+        expression: str
+            The value expression of the variable."""
+        if isinstance(expression, str):
+            self._expression = expression
+            self._application[self.name] = expression
+        else:
+            self._application.logger.error("Expression must be a string")
+
+    @property
+    def unit_system(self):
+        """Unit system of the expression as a string."""
+        return self._variable.unit_system
+
+    @property
+    def units(self):
+        """Units."""
+        return self._variable.units
+
+    @property
+    def value(self):
+        """Value."""
+        return self._variable.value
+
+    @property
+    def numeric_value(self):
+        """Numeric part of the expression as a float value."""
+        return self._variable.numeric_value
+
+    @property
+    def string_value(self):
+        """Return a string which combine the numeric_value and the units."""
+        return self._variable.string_value
+
+    @pyaedt_function_handler()
+    def hide_variable(self, value=True):
+        """Set the variable to a hidden variable.
+
+        Parameters
+        ----------
+        value : bool, optional
+            Whether the variable is a hidden variable. The default is ``True``.
+
+        Returns
+        bool
+        """
+        self._application.variable_manager[self._name].hidden = value
+        return True
+
+    @pyaedt_function_handler()
+    def read_only_variable(self, value=True):
+        """Set the variable to a read only variable.
+
+        Parameters
+        ----------
+        value : bool, optional
+            Whether the variable is a read only variable. The default is ``True``.
+
+        Returns
+        -------
+        bool
+        """
+        self._application.variable_manager[self._name].read_only = value
+        return True
+
+
+class Layer3D(object):
+    """Layer3D Class member of Stackup3D."""
```

Comment:
The description should start with "Provides ...." so that it appears correctly in the summary table.